### ISSUE TITLE
D-ITG generated script

### ISSUE THREAD

**Issue by ghalwash** [devenbansod/pcap_to_ditg#5]
Hi,
I am trying to use the generated Script files for my D-ITG nodes. The script for D-ITG looks a little bit different as shown in the link http://traffic.comics.unina.it/software/ITG/manual/. 

I have now files that are *.ps/.itds/.ditg, I really do not know how to use them in mininet topology and DITG hosts.

thanks

**Comment by devenbansod** [comment]
Hi @ghalwash 

1. Run the script for generating the `list_file`. I used [wrccdc2012.pcap.gz](https://github.com/devenbansod/pcap_to_ditg/files/2590032/wrccdc2012.pcap.gz) for doing this example.
```
python run.py -p wrccdc2012.pcap mapper_file list_file > list_file
```
This generates a `list_file` with 57 IPs mentioned in it.

2. Fill the mapper_file with the mappings you want to do. Mine (for this example) looks like:
```
Host,Start_row,End_row,Number,IP
h1,1,28,28,10.0.0.1
h2,29,57,29,10.0.0.2
```

3. Now, you're ready to run the main script:
```
python run.py wrccdc2012.pcap mapper_file list_file
```
This will generate one folder for each of the mapped host. For this example, 
[10.0.1.10_ditg_files.zip](https://github.com/devenbansod/pcap_to_ditg/files/2590044/10.0.1.10_ditg_files.zip)
[10.0.1.20_ditg_files.zip](https://github.com/devenbansod/pcap_to_ditg/files/2590045/10.0.1.20_ditg_files.zip)

4. These folders should be put on the respective host (supposedly in your mininet topology).


This is what the generated`.ditg` file looks for me:
```
-z 3 -a 10.0.1.10  -Fs 10.0.10.248_74.125.45.125.ps  -Ft 10.0.10.248_74.125.45.125.idts -T TCP
-z 4 -a 10.0.1.10  -Fs 10.1.1.47_10.0.10.248.ps  -Ft 10.1.1.47_10.0.10.248.idts -T TCP
-z 5 -a 10.0.1.10  -Fs 192.168.250.80_10.0.10.69.ps  -Ft 192.168.250.80_10.0.10.69.idts -T TCP
-z 4 -a 10.0.1.10  -Fs 10.1.1.47_10.0.10.37.ps  -Ft 10.1.1.47_10.0.10.37.idts -T TCP
```

5. Start `./ITGRecv` on both the hosts to make sure they are setup to receive the packets
```
./ITGRecv
```

6. Use the generated script file `.ditg` to run the `ITGSend`
```
./ITGSend 10.0.1.10.ditg
```

This should start the traffic re-run.

Please let me know if you see any error while doing these steps.

**Comment by ghalwash** [comment]
Hi @devenbansod 

I tried the script.
I noticed that I need to have the file "10.0.1.10.ditg" in the same folder of the ITGSend script file. If used a path to the file, for example 
` ./ITGSend ./10.0.1.10_ditg_files/10.0.1.10.ditg
` 
I get the following error
```
ITGSend version 2.8.1 (r1023)
Compile-time options: bursty multiport
 ** ERROR **  Flow 1 aborted:  flowParser : Error in opening the file of the packets sizes 
 Type ./ITGSend -h for help 
 ** ERROR **  Flow 2 aborted:  flowParser : Error in opening the file of the packets sizes 
 Type ./ITGSend -h for help 
 ** ERROR **  Flow 3 aborted:  flowParser : Error in opening the file of the packets sizes 
 Type ./ITGSend -h for help 

```
So I simply but the content of the folder with the ./ITGSend script in the same folder (unziped the file and but the content in the folder containing the ./ITGSend)

my steps are as follow:
1- I opened an Xterm for h1 and h2 from mininet terminal
2- run the ./ITGRecv on both
3- run ./ITGSend 10.0.1.10.ditg

Now I am having an error of 
```
ITGSend version 2.8.1 (r1023)
Compile-time options: bursty multiport
Connect error in createTransportChan(): No route to host
```
I checked the connectivity between both h1 and h2 (ping 10.0.0.1 & ping 10.0.0.2)

I tested another ./ITGSend for each device, and the connection was OK
```
./ITGSend -a 10.0.0.1
./ITGSend -a 10.0.0.2 
```
I am also confused about how the .ditg file looks like, as the destination addresses are 10.0.1.10 and 10.0.1.20, which I was expecting to be the host IP of h1 (10.0.0.1) or the host IP of h2 (10.0.0.2)

**Comment by devenbansod** [comment]
@ghalwash thanks for sharing your findings.
 
1. You're right that `ITGSend` needs to be present inside the generated directory

2. For the other issue with incorrect IPs (different from those specified in `mapper_file`):
I think the issue is you're using `run.py` from latest master, but the installed version of the library `pcap_to_ditg` is older.

I've just published a newer version `v1.0.1` (I sort of screwed up `v1.0.0` 👎 ) at https://github.com/devenbansod/pcap_to_ditg/releases/tag/v1.0.1 

I've also published it at https://pypi.org/project/pcap_to_ditg. So you could do:
```sh
pip uninstall pcap_to_ditg
pip install pcap_to_ditg
```

and try out with the latest version.

Let me know if you face any issues.

**Comment by ghalwash** [comment]
Great!
that solved the Issue
Thank you so much
 :+1:
